includes/class-scd-ext-access-control.php

```diff
@@ -229,15 +233,19 @@ public function is_dynamic_drip_type_content_blocked( $lesson_id ) {
 		}
 
 		$lesson_becomes_available_date = $this->get_lesson_drip_date( $lesson_id , $user_id );
-		$today                         = new DateTime(); // Get today's date
+		$today                         = Sensei_Content_Drip()->utils->current_datetime()->setTime( 0, 0, 0 );
+
+		if ( ! $lesson_becomes_available_date ) {
```

Comment:
`get_lesson_drip_date` has always been able to return `false` but we weren't handling it in all of its uses. It might have been hidden by the old comparison, but now since I'm treating it as a `DateTimeImmutable` I handle that return explicitly.